pyanaconda/kickstart_dispatcher.py

```diff
@@ -0,0 +1,355 @@
+                if commands is None:
+                    continue
+                if commands == [] or name in commands:
+                    filtered_elements.append(element)
+        return filtered_elements
+
+    @staticmethod
+    def get_kickstart_from_elements(elements=None):
+        """Returns kickstart generated from elements.
+
+        :param elements: list of kickstart elements for generated kickstart
+        :type elements: list(KickstartElement)
+
+        :return: kickstart
+        :rtype: str
+        """
+        return "".join(element.content for element in elements)
+
+    def __str__(self):
+        return str(self._elements)
+
+
+class TrackedKickstartElements(KickstartElements):
+    """Container for kickstart elements with tracking."""
+    def __init__(self):
+        super().__init__()
+        self._processed_elements = set()
+
+    def get_elements(self, commands=None, sections=None, addons=None, process=False):
+        """Returns selected elements and tracks them as processed.
+
+        :param commands: names of commands to be returned or [] for all commands
+        :type commands: list
+        :param sections: names of sections to be returned or [] for all sections
+        :type sections: list
+        :param addons: names of addons to be returned or [] for all addons
+        :type addons: list
+        :param process: track returned elements as processed
+        :type process: bool
+
+        :return: list of filtered elements
+        :rtype: list(KickstartElement)
+        """
+        elements = super().get_elements(commands, sections, addons)
+        if process:
+            self._processed_elements.update(elements)
+        return elements
+
+    @property
+    def unprocessed_elements(self):
+        """List of all elements not tracked as processed.
+
+        :return: list of all elements not tracked as processed
+        :rtype: list(KickstartElement)
+        """
+        return [element for element in self._elements
+                if element not in self._processed_elements]
+
+
+class StoreSection(Section):
+    """Section for storing section content and header line references.
+
+    Similarly as NullSection defines a section that parser will recognize (ie
+    will not raise an error). The section will pass itself to an object for storing
+    sections if supplied.
+    """
+
+    allLines = True
+
+    def __init__(self, *args, **kwargs):
+        """Create a new StoreSection instance.
+
+        You must pass a sectionOpen parameter (including a leading '%') for the
+        section to make it valid but just ignored. If you want to store the
+        content, supply a store argument.
+
+        Required kwargs:
+        sectionOpen - section name, including '%' starting character
+
+        Optional kwargs:
+        store - an instance of an object for storing the section
+                (SplitKickstartParser) which must provide
+                add_section(StoreSection) method
```

Comment:
Please merge this to two lines instead of three. I guess it should be doable in 99 characters limit.